contrib/net/http/trace.go

```diff
@@ -63,24 +77,23 @@ func TraceAndServe(h http.Handler, w http.ResponseWriter, r *http.Request, cfg *
 	if spanctx, err := tracer.Extract(tracer.HTTPHeadersCarrier(r.Header)); err == nil {
 		opts = append(opts, tracer.ChildOf(spanctx))
 	}
-	span, ctx := tracer.StartSpanFromContext(r.Context(), "http.request", opts...)
-	rw, ddrw := wrapResponseWriter(w)
-	defer func() {
-		if ddrw.status == 0 {
-			span.SetTag(ext.HTTPCode, "200")
-		} else {
-			span.SetTag(ext.HTTPCode, strconv.Itoa(ddrw.status))
-		}
-		if ddrw.status >= 500 && ddrw.status < 600 {
-			span.SetTag(ext.Error, fmt.Errorf("%d: %s", ddrw.status, http.StatusText(ddrw.status)))
-		}
-		span.Finish(cfg.FinishOpts...)
-	}()
+	return tracer.StartSpanFromContext(r.Context(), "http.request", opts...)
+}
 
-	if appsec.Enabled() {
-		h = httpsec.WrapHandler(h, span, cfg.RouteParams)
+// FinishRequestSpan finishes the given HTTP request span with the standard list of HTTP request span tags.
+// Any further span finish option can be added with opts.
+func FinishRequestSpan(s tracer.Span, status int, opts ...tracer.FinishOption) {
+	var statusStr string
+	if status == 0 {
+		statusStr = "200"
+	} else {
+		statusStr = strconv.Itoa(status)
 	}
-	h.ServeHTTP(rw, r.WithContext(ctx))
+	s.SetTag(ext.HTTPCode, statusStr)
```

Comment:
nit: I preferred the way the code was previously where `s.SetTag` is within the if statement. It was a bit clearer to read that way in my opinion.

Reply:
I was about to restore it until I realized we were converting the status code into a string twice in that function.
So I reused `statusStr` in the second place in https://github.com/DataDog/dd-trace-go/pull/1286/commits/41f4c74ae4dfd7847f14144baf18819f8abeb198

WDYT?

Reply:
Ah yep that's fine then 👍